homeassistant/components/comelit/__init__.py

```diff
@@ -0,0 +1,53 @@
+"""Comelit integration."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PIN,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
+from homeassistant.core import Event, HomeAssistant
+
+from .const import _LOGGER, DOMAIN
+from .coordinator import ComelitSerialBridge
+
+PLATFORMS = [Platform.LIGHT]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Comelit platform."""
+    _LOGGER.debug("Setting up Comelit component")
+    coordinator = ComelitSerialBridge(
+        entry.data[CONF_HOST],
+        entry.data[CONF_PIN],
+        hass=hass,
+    )
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+
+    async def async_close_connection(event: Event) -> None:
+        """Close Comelit connection on HA Stop."""
+        await coordinator.api.logout()
+
+    entry.async_on_unload(
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_close_connection)
+    )
```

Comment:
Why is this needed? We normally don't bother with tearing down things when Home Assistant stops.

Reply:
Took inspiration from another integration, removed.